crates/burn-jit/src/kernel/conv/conv2d/implicit_gemm.rs

```diff
@@ -0,0 +1,343 @@
+        options.padding[1],
+        options.dilation[1],
+        width,
+    );
+
+    if !can_do_implicit_gemm(&input, &weight, &options, out_h, out_w) {
+        panic!(
+            "Requirements for implicit GEMM not met:
+- CMMA must be available
+- `batch_size * out_h * out_w` must be divisible by 16
+- `out_channels` must be divisible by 16
+- `in_channels * kernel_h * kernel_w` must be divisible by 16
+- `groups` must be 1
+        "
+        );
+    }
+
+    let out_shape = Shape::new([batch_size, out_h, out_w, out_channels]);
+    let mut out = empty_device(input.client.clone(), input.device.clone(), out_shape);
+
+    // Implicit GEMM matrix size
+    let gemm_m = (batch_size * out_h * out_w) as u32;
+    let gemm_n = out_channels as u32;
+    let gemm_k = in_channels * kernel_h * kernel_w;
+    let slice_size = kernel_h * kernel_w * in_channels;
+
+    // `CUBE_DIM_X` must be a multiple of `WARP_SIZE`
+    // 128x2 means we have 8 warps and a cube computes a 32x64 output tile
+    let cube_dim = CubeDim {
+        x: CUBE_DIM_X,
+        y: CUBE_DIM_Y,
+        z: 1,
+    };
+
+    let cube_count_x = gemm_m.div_ceil(CMMA_M * CUBE_DIM_X / WARP_SIZE);
+    let cube_count_y = gemm_n.div_ceil(CMMA_N * CUBE_DIM_Y);
+
+    let cube_count = CubeCount::Static(cube_count_x, cube_count_y, 1);
+
+    unsafe {
+        implicit_gemm_kernel::launch_unchecked::<F, R>(
+            &input.client,
+            cube_count,
+            cube_dim,
+            input.as_tensor_arg(1),
+            weight.as_tensor_arg(1),
+            out.as_tensor_arg(1),
+            ScalarArg::new(gemm_m),
+            ScalarArg::new(gemm_n),
+            ScalarArg::new(gemm_k as u32),
+            ScalarArg::new(slice_size as u32),
+            KernelArgsLaunch::new(
+                ScalarArg::new(options.stride[0] as u32),
+                ScalarArg::new(options.stride[1] as u32),
+                ScalarArg::new(options.padding[0] as i32),
+                ScalarArg::new(options.padding[1] as i32),
+                ScalarArg::new(options.dilation[0] as u32),
+                ScalarArg::new(options.dilation[1] as u32),
+            ),
+            kernel_h as u32,
+            kernel_w as u32,
+        )
+    };
+
+    if let Some(bias) = bias {
+        let bias = reshape(bias, Shape::new([1, 1, 1, out_channels]));
+        out = JitBackend::<R, F, I>::float_add(out, bias);
+    }
+
+    permute(out, [0, 3, 1, 2])
+}
+
+#[derive(CubeLaunch)]
+struct KernelArgs {
+    stride_h: u32,
+    stride_w: u32,
+    pad_h: i32,
+    pad_w: i32,
+    dilation_h: u32,
+    dilation_w: u32,
+}
+
+#[cube(launch_unchecked)]
+fn implicit_gemm_kernel<F: Float>(
+    input: &Tensor<F>,
+    weight: &Tensor<F>,
+    out: &mut Tensor<F>,
+    gemm_m: u32,
+    gemm_n: u32,
+    gemm_k: u32,
+    slice_size: u32,
+    args: &KernelArgs,
+    #[comptime] kernel_h: u32,
+    #[comptime] kernel_w: u32,
+) {
+    let kernel_size = comptime! { kernel_h * kernel_w };
```

Comment:
Doesn't arithmetic on two comptime values always give a comptime? Why the need for a comptime macro?

Reply:
That was left over from a refactor, I'll just remove it.